parcels/field.py

```diff
@@ -826,6 +839,73 @@ def ccode_eval(self, var, t, z, y, x):
     def ccode_convert(self, _, z, y, x):
         return self.units.ccode_to_target(x, y, z)
 
+    #def find_in_block(self, chunksize, ti, zi, yi, xi):
+    #    #index = (ti, zi, yi, xi)
+    #    index = (ti, yi, xi)
+    #    block = tuple((int(index[i]/chunksize[i]) for i in range(len(index))))
+    #    local_index = tuple(index[i] - chunksize[i]*block[i] for i in range(len(index)))
+    #    return block, local_index
+
+    def get_block_id(self, block):
+        return np.ravel_multi_index(block, self.nchunks)
+
+    def get_block(self, bid):
+        return np.unravel_index(bid, self.nchunks)
+
+    def chunk_data(self):
+        if isinstance(self.data, da.core.Array):
+            if len(self.data.chunks[0]) > 1:
+                self.data = self.data.rechunk(sum(((self.grid.tdim,), self.data.chunksize[1:]), ()))
+            npartitions = self.data.npartitions
+            chunksize = self.data.chunksize
+            chunks = self.data.chunks
+        else:
+            npartitions = 1
+            chunksize = self.data.shape
+            chunks = tuple((t,) for t in chunksize)
+
+        self.nchunks = tuple(len(chunks[i]) for i in range(len(chunks)))
```

Comment:
In the case of dask arrays:
```suggestion
        self.nchunks = self.data.numblocks
```
and `self.nchunks = (1,) * len(chunksize)` or something to that effect for the non-dask case.